cluster/evaluator/evaluator_ringmaster_wrapper.py

```diff
@@ -0,0 +1,225 @@
+]
+'''
+
+Model = namedtuple("Model", ["hash", "name", "model_pb", "flags"])
+
+def setup_ringmaster(model_a, model_b, start_time, num_games):
+    with open(CTL_FILENAME, "w") as ctl_f:
+        ctl_f.write(CTL_FILE.format(
+            m_a=model_a,
+            m_b=model_b,
+            start_time=start_time,
+            num_games=num_games))
+
+
+def call_ringmaster(num_games):
+    process = subprocess.run(
+        ["ringmaster", CTL_FILENAME, "run"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        timeout=num_games*20*60)
+    # log errors somewhere
+    if process.returncode != 0:
+        print("Error return=", process.returncode)
+        print(process.stdout)
+        print(process.stderr)
+
+    return process.returncode == 0
+
+
+def copy_to_gcs(src, dst):
+    assert gfile.Exists(src), src
+    assert not gfile.Exists(dst), dst
+
+    print("Saving to", dst)
+    with gfile.GFile(src, "rb") as src_f, gfile.GFile(dst, "wb") as dst_f:
+        shutil.copyfileobj(src_f, dst_f)
+
+
+def record_results(bt_table, sgf_base, num_games, start_time):
+    games = os.listdir(CTL_GAME_DIR)
+    failed = len(games) != num_games
+
+    if failed:
+        # Upload something? log error somewhere?
+        assert False, (len(games), num_games)
+
+    # Upload .log and .report along side all .games
+    copy_to_gcs(CTL_LOG, os.path.join(sgf_base, CTL_LOG))
+    copy_to_gcs(CTL_REPORT, os.path.join(sgf_base, CTL_REPORT))
+
+    rows = []
+    for game_fn in games:
+        game_path = os.path.join(CTL_GAME_DIR, game_fn)
+        copy_to_gcs(game_path, os.path.join(sgf_base, game_fn))
+
+        metadata = bigtable_output.process_game(game_path)
+        metadata["sgf"] = game_fn
+        metadata["tool"] = "evaluator_ringmaster"
+
+        # game_fn contains timestamp and is hence unique.
+        row = bt_table.row(game_fn)
+        for column, value in metadata.items():
+            row.set_cell(METADATA, column, value)
+
+        rows.append(row)
+
+    response = bt_table.mutate_rows(rows)
+
+    # validate that all rows were written successfully
+    all_good = True
+    for i, status in enumerate(response):
+        if status.code is not 0:
+            print("Row number {} failed to write {}".format(i, status))
+            all_good = False
+
+    return all_good
+
+
+def get_cbt_model(bt_table, model_hash):
+    model_row = bt_table.read_row(MODEL_ROW_FMT.format(model_hash))
+    def get_cell(cell):
+        return model_row.cell_value(METADATA, cell.encode()).decode()
+
+    model_flags = get_cell("model_flags")
+    model_pb = get_cell("model_path")
+    model_name = get_cell("model")
+    return Model(
+        model_hash,
+        model_name,
+        os.path.basename(model_pb),
+        model_flags.replace("flags: ", ""),
+    )
+
+
+if __name__ == "__main__":
+    ENV_VARS = [
+        "PROJECT",
+        "CBT_INSTANCE",
+        "CBT_TABLE",
+        "MODEL_WHITE",
```

Comment:
MODEL_WHITE and MODEL_BLACK aren't good names if ringmaster is going to be in charge of alternating colors.  eval.cc is going towards "model_eval" and "model_target"... this could also just be "model a" and "model b" though, as the expectation is that white/black is handled by ringmaster